api/incubator/src/test/java/io/opentelemetry/api/incubator/config/InstrumentationConfigUtilTest.java

```diff
@@ -0,0 +1,169 @@
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfiguration;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+class InstrumentationConfigUtilTest {
+
+  /**
+   * See <a
+   * href="https://github.com/open-telemetry/opentelemetry-configuration/blob/main/examples/kitchen-sink.yaml">kitchen-sink.yaml</a>.
+   */
+  private static final String kitchenSinkInstrumentationConfig =
+      "instrumentation:\n"
+          + "  general:\n"
+          + "    peer:\n"
+          + "      service_mapping:\n"
+          + "        - peer: 1.2.3.4\n"
+          + "          service: FooService\n"
+          + "        - peer: 2.3.4.5\n"
+          + "          service: BarService\n"
+          + "    http:\n"
+          + "      client:\n"
+          + "        request_captured_headers:\n"
+          + "          - client-request-header1\n"
+          + "          - client-request-header2\n"
+          + "        response_captured_headers:\n"
+          + "          - client-response-header1\n"
+          + "          - client-response-header2\n"
+          + "      server:\n"
+          + "        request_captured_headers:\n"
+          + "          - server-request-header1\n"
+          + "          - server-request-header2\n"
+          + "        response_captured_headers:\n"
+          + "          - server-response-header1\n"
+          + "          - server-response-header2\n"
+          + "  java:\n"
+          + "    example:\n"
+          + "      property: \"value\"";
+
+  private static final ConfigProvider kitchenSinkConfigProvider =
+      toConfigProvider(kitchenSinkInstrumentationConfig);
+  private static final ConfigProvider emptyInstrumentationConfigProvider =
+      toConfigProvider("instrumentation:\n");
+  private static final ConfigProvider emptyGeneralConfigProvider =
+      toConfigProvider("instrumentation:\n  general:\n");
+  private static final ConfigProvider emptyHttpConfigProvider =
+      toConfigProvider("instrumentation:\n  general:\n    http:\n");
+
+  private static ConfigProvider toConfigProvider(String configYaml) {
+    OpenTelemetryConfiguration configuration =
+        FileConfiguration.parse(
+            new ByteArrayInputStream(configYaml.getBytes(StandardCharsets.UTF_8)));
+    return FileConfigProvider.create(configuration);
+  }
+
+  @Test
+  void peerServiceMapping() {
+    assertThat(InstrumentationConfigUtil.peerServiceMapping(kitchenSinkConfigProvider))
+        .isEqualTo(ImmutableMap.of("1.2.3.4", "FooService", "2.3.4.5", "BarService"));
+    assertThat(InstrumentationConfigUtil.peerServiceMapping(emptyInstrumentationConfigProvider))
+        .isNull();
+    assertThat(InstrumentationConfigUtil.peerServiceMapping(emptyGeneralConfigProvider)).isNull();
+    assertThat(InstrumentationConfigUtil.peerServiceMapping(emptyHttpConfigProvider)).isNull();
+  }
+
+  @Test
```

Comment:
recommend tests for below cases possibly something that can be refactored to a common fixture for all, but even if explicit 1. path is incomplete 2. middle of the path is the wrong type 3. end of the path is the wrong type 4. end of the path is empty